systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java

```diff
@@ -412,7 +412,15 @@ protected void setupEnvAndUpgradeClusterOperator(ExtensionContext extensionConte
         if (!cmdKubeClient().getResources(getResourceApiVersion(Kafka.RESOURCE_PLURAL, operatorVersion)).contains(clusterName)) {
             // Deploy a Kafka cluster
             if ("HEAD".equals(testParameters.getString("fromVersion"))) {
-                resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, 3, 3).build());
+                resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, 3, 3)
+                    .editSpec()
+                        .editKafka()
+                            .withVersion(kafkaVersion)
+                            .addToConfig("log.message.format.version", kafkaVersion.substring(0, 3))
+                            .addToConfig("inter.broker.protocol.version", kafkaVersion.substring(0, 3))
```

Comment:
I wonder if you should use some smarter way than `substring(0, 3)` to extract the format. While it is a long time before we have a minor version with more digits, one day it might happen.

Reply:
Maybe get it from `kafka-versions.yaml` is better